csrc/device_lower/pass/double_buffer.cpp

```diff
@@ -289,11 +387,215 @@ class DoubleBufferLoopCloner : public kir::IrVisitor {
           NVF_ERROR(double_buffer_tv != nullptr);
           return out_tv == double_buffer_tv;
         });
-    if ((loop_type_ == DoubleBufferLoopStage::Prolog &&
-         is_double_buffer_load_expr) ||
-        (loop_type_ == DoubleBufferLoopStage::Epilog &&
-         !is_double_buffer_load_expr)) {
-      cloned_scopes_.back()->push_back(expr);
+
+    if (has_cp_async_bulk_) {
+      auto gpu_lower = GpuLower::current();
+
+      // Expr that is a part of cpAsyncBulk synchronization proces, added
+      //  ealier to satisfy other passes checks, are handled already won't
+      //  be pushed to the new scope.
+      // There can be cases where cpAsyncBulk data moevement is not a part of
```

Comment:
```suggestion
      // There can be cases where cpAsyncBulk data movement is not a part of
```

Reply:
Fixed along the rebase - thanks for pointing this out!